### PR TITLE
added roles endpoint

### PR DESCRIPTION
**Pull request by matmccabe**
Added roles endpoint, need review and some syntax troubleshooting for .rst file

### CODE REVIEW

**Comment by MrGeneration** [comment]
Thank you for your contribution and your time.
Just to clarify. Is this a WIP PR or are you waiting for actual QA?

I've had a 10 minute look and fixed mayor issues. Already.
However, the code blocks cannot be lexed without enforcing as JSON (this is due to us providing a comment with the return code). Some of your return codes bring 4xx codes which doesn't look right. At least not if this is going to show up in the documentation. 

Some payloads send to the API and their responses look mixed up as well.
At least I can't follow them - I'm not sure what happened there. 🙏

**Comment by ralf401** [comment]
Hi @matmccabe 
thanks a lot for your contribution! I redacted your draft and commited it. 

@mantas can you have a look if ok? Thank you!

File: api/roles.rst [hunk 1]
```diff
@@ -0,0 +1,751 @@
...
+
+   # HTTP-Code 200 OK
+
+   {
+      "record_ids": [
+         1,
+         2,
+         3
+      ],
+      "assets": {
```
Comment (mantas) [inline api/roles.rst:24]: Lets make this more readable by not including `full=true` flag and `assets` content.

File: api/roles.rst
Comment (mantas) [inline api/roles.rst:1]: Minor nitpick, but other file names in this folder are singular :)

File: api/roles.rst (hunk 1 above)
Comment (ralf401) [reply api/roles.rst:24]: Oh I guess it was not really correct at all in my first iteration ... Changed the request to the "standard" `/api/v1/roles` now so it should look better.

File: index.rst [hunk 2]
```diff
@@ -81,8 +81,10 @@ The Zammad documentation consists of three parts:
 
    /api/intro
    /api/user
+   /api/roles
    /api/organization
    /api/group
+   /api/roles
```
Comment (mantas) [inline index.rst:87]: I guess one of these `/api/roles` mentions is not needed?

**Pull request merged**